apps/trading/client-pages/market/market-selector-item.spec.tsx

```diff
@@ -17,14 +17,29 @@ import {
 } from '@vegaprotocol/types';
 import { addDecimalsFormatNumber } from '@vegaprotocol/utils';
 
+jest.mock('@vegaprotocol/markets', () => ({
```

Comment:
There is a MockedProvider set up for this test. We can gain more value from this test if you create a mock for the query used in marketCandlesProvider and pass it to the mocks array

Reply:
I see, I mocked useCandles hook here, and added tests to useCandles. 

I had some trouble trying to mock this, it expects data undefined on mocked response, not sure what I'm missing. 
<img width="590" alt="image" src="https://github.com/vegaprotocol/frontend-monorepo/assets/16125548/2890c0ac-5c6e-4daf-896a-afd619c753b3">

Reply:
@MadalinaRaicu for the query to successfully resolve within the test you need to ensure that variables (and return data) defined in the mock _exactly_ match what is provided to the query in the code. 

Here, it looks like you might have modified what variables get passed to the data provider, but internally the data provider will still be calling the _actual query_ with `marketId`, `interval` and `since`, so you'll need to update the `MockedResponse` to use the correct variables.

Have a read here: https://www.apollographql.com/docs/react/development-testing/testing/#defining-mocked-responses

Reply:
I have tried to mock both the query and the subscription from marketCandlesDataProvider 

<img width="750" alt="image" src="https://github.com/vegaprotocol/frontend-monorepo/assets/16125548/a5ccb50f-7870-4d08-9531-2103a88d4b7a">


but since this test is from trading 
I can't manage to not mock useCandles hook which is part of `@vegaprotocol/markets` lib 
so the 24h volume would be 0.00

Reply:
Okay two things 

1. The variables and return type must exactly match. The interval variable here is 1D when actually the query uses 1H, so that would've resulted in the mock query not 'matching' the query used in the test.
2. Any query which uses a date as a variable can be tricky as the date timestamp must match between the mock and what actually runs in the test, so often its necessary to mock the date constructor so that there isn't a delay between when the test runner creates the date and when the code under test creates the date. 

Have a look at the latest commit. I've added a mock for the date creation to ensure whenever you call Date.now you get the same date. Secondly I've updated the variables so that hourly intervals are used.

We now have the added benefit that we are
A) also inherently testing the useCandles hook
B) also testing the date provider